scripts/liveValidation.js

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License in the project root for license information.
+
+'use strict';
+
+const utils = require('../test/util/utils'),
+    request = require('request-promise-native'),
+    zlib = require('zlib');
+
+const repoUrl = utils.getRepoUrl(),
+    validationService = "https://app.azure-devex-tools.com/api/validations",
+    branch = utils.getSourceBranch(),
+    processingDelay = 20,
+    isRunningInTraviCI = process.env.MODE === 'liveValidation' && process.env.PR_ONLY === 'true',
+    specsPaths = utils.getFilesChangedInPR(),
+    regex = /resource-manager[\\|\/](.*?)[\\|\/].*?[\\|\/](.*?)[\\|\/]/,
+    successThreshold = 90,
+    validationModels = new Map();
+
+let durationInSeconds = parseInt(process.env.LIVE_VALIDATION_DURATION_IN_MINUTES) * 60;
+if (isNaN(durationInSeconds)) {
+    durationInSeconds = 180;
+}
+
+async function runScript() {
+    // See whether script is in Travis CI context
+    console.log(`isRunningInTraviCI: ${isRunningInTraviCI}`);
+    for (const specPath of specsPaths) {
+        let matchResult = specPath.match(regex);
+
+        if (matchResult === null) {
+            continue;
+        }
+
+        let resourceProvider = matchResult[1];
+        let apiVersion = matchResult[2];
+
+        if (!validationModels.has(resourceProvider)) {
+            validationModels.set(resourceProvider, new Set());
+        }
+
+        validationModels.get(resourceProvider).add(apiVersion);
+    }
+
+    if (validationModels.size === 0) {
+        console.log("Change didn't affect any swagger specs. No validation to be done.");
+        return;
+    }
+
+    let resourceProvider = validationModels.keys().next().value;
```

Comment:
could this fail? is there always a keys().next().value?

Reply:
are you only expecting 1 resourceProvider?

Reply:
same as for api version

---
In reply to: [159793672](https://github.com/Azure/azure-rest-api-specs/pull/2211#discussion_r159793672) [](ancestors = 159793672)

Reply:
here it won't fail since we just return before that if the validationModels is empty (i.e. there is no swagger in the change)